### DISCUSSION TITLE
problems moving objects

### DISCUSSION THREAD

**Discussion by rossmarinaro** [enable3d/enable3d#324] [General]
I've been using enable3d for a while now, and I have to say i'm a fan. However, one thing that absolutely needs to change and be improved on is the ability to easily just move an object. I know this sounds dumb, and a presumably simple thing to accomplish, but it's really not. Ive been pulling my hair out just trying to get a child of a glb to move independently with different collision flags. It seems like nothing is making this platform move, and certain methods like set, or copy on position just dont work, throw errors, and ultimately don't budge. really sick of sitting here trial and error hoping this platform will move. The object tweens, just not the body no matter what flag i set. Really not practical at all when Im sitting for hours with no results trying to get a box to move.... I even added another extended3d object and copied the position, although Im starting to think that any solutuon I find just isnt a solution or yield any desirable result. Like, Im just trying to make a box move and its a hug problem. Not a fan of this at all.

**Comment by rossmarinaro** [comment]
update: here is how i added a moving platform, basically an invisible separate box in the same location, and I just disabled the body of the original problem child.

       if (child.name === 'platform')
          {

            let tmp = child.position.clone(),
                platform = new ExtendedObject3D();

             platform.position.copy(tmp);

            this.scene.third.physics.add.existing(platform, {
              shape: 'box', 
              width: 90,
              height: 5,
              depth: 90,
              mass: 0,
              collisionFlags: 2
            });
 
            this.scene.tweens.add({
              targets: tmp, 
              duration: 5000, 
              repeatDelay: 3000, 
              delay: 3000, 
              hold: 3000,
              ease: 'Sine.easeInOut', 
              y: tmp.y + 70, 
              repeat: -1, 
              yoyo: true,
              onUpdate: ()=> {
                child.position.setY(tmp.y); 
                platform.position.setY(tmp.y);
                platform.body.needUpdate = true;
              }
            });

            if (child.body !== null && child.body !== undefined)
              this.scene.third.physics.destroy(child)
          }

**Comment by yandeu** [comment]
Can you please share some visuals of what you are trying to accomplish?

I find movements using ammo.js very simple. Just like in other physic engines.

**Reply by rossmarinaro** [reply]
![Capture](https://user-images.githubusercontent.com/44373734/192341080-358080db-5377-4c96-8f88-1e27493993b8.PNG)

I had to make another platform body around the object because the one which is green won't budge. (it's part of the map)
